File: pigatize.rb
VOWELS = ['a', 'e', 'i', 'o', 'u']

y_not = rand(1..2)

if y_not == 2
  VOWELS << 'y'
end

def pigatize(text)
  if starts_with_vowel?(text)
    puts text + "way"
  else
    text_arr = text.chars
    pop = text_arr.shift
    text_arr << pop
    puts text_arr.join('') + "ay"
  end
end

def starts_with_vowel?(text)
  if VOWELS.include?(text.chars.first)
    true
  end
end


pigatize("alphabet")
pigatize("garrett")
pigatize("illinoise")
pigatize("regency")
pigatize("yokel")

File: pigatize.js
function pigatize(text) {
    var pop = text.slice(0,1);
    if ( vowelCheck(text) === true ) console.log(text + 'way');
    else  console.log(text.slice(1) + pop + 'ay');
}

var vowelCheck = function (text) {
  if ( text[0].search(/[aeiouAEIOU]/) === 0 ) return true;
  else return false;
};

pigatize("alphabet");
pigatize("germanic");
pigatize("oligarchy");
pigatize("populations");
